go.mod

```diff
@@ -265,3 +261,9 @@ replace (
 	k8s.io/kubernetes => k8s.io/kubernetes v0.29.3
 	k8s.io/mount-utils => k8s.io/mount-utils v0.29.3
 )
+
+// There is an error with newer versions of prometheus wrt to control-plane-pki-operator
+replace (
+	github.com/prometheus/client_golang => github.com/prometheus/client_golang v1.18.0
+	github.com/prometheus/common => github.com/prometheus/common v0.47.0
+)
```

Comment:
I am unsure what the error is 😕

Reply:
```
% make build
CGO_ENABLED=1 GO111MODULE=on GOFLAGS=-mod=vendor go build -gcflags=all='-N -l' -o bin/hypershift-operator ./hypershift-operator
CGO_ENABLED=1 GO111MODULE=on GOFLAGS=-mod=vendor go build -gcflags=all='-N -l' -o bin/control-plane-operator ./control-plane-operator
CGO_ENABLED=1 GO111MODULE=on GOFLAGS=-mod=vendor go build -gcflags=all='-N -l' -o bin/control-plane-pki-operator ./control-plane-pki-operator
# github.com/prometheus/client_golang/prometheus/testutil/promlint
vendor/github.com/prometheus/client_golang/prometheus/testutil/promlint/promlint.go:71:38: undefined: expfmt.NewFormat
make: *** [control-plane-pki-operator] Error 1
```

Reply:
https://github.com/prometheus/client_golang/releases/tag/v1.19.0

Reply:
I also pinned Prometheus deps in CAPA https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/main/go.mod#L7-L10 because `component-base` still using old versions

Reply:
Ah I see. Steve was discussing this with me as well on Slack.